src/PrestaShopBundle/Controller/Admin/Improve/Design/CmsPageController.php

```diff
@@ -249,12 +253,78 @@ public function createAction(Request $request)
     public function editAction(Request $request, $cmsPageId)
     {
         $cmsPageId = (int) $cmsPageId;
-
+        $context = $this->getContext();
+        $shopContext = $context->shop->getContextType();
+        $db = Db::getInstance();
+        $prefix = $db->getPrefix();
+        $isSame = $db->getRow('select ' . CmsPageQueryBuilder::getIsSameQuery($prefix) . ' from ' . $prefix . 'cms c where c.id_cms = ' . $cmsPageId);
         try {
+            $forceEditUrl = $forceEditMessage = '';
+            if (Shop::isFeatureActive()) {
```

Comment:
May be move this code to a query handler would be better ?

Reply:
(and testable)

Reply:
You mean `$db->getRow()` ? I agree, but for now, I'm a symfony novice, my knowledge is insufficient for this.
I'm ready to learn, but I don't know where to start.

Reply:
Testable: Sure, I already wrote tests, but all PS testing tools are completely new-land for me. I'm not able to write a test for this for the moment.